cin_validator/rules/cin2022_23/rule_8816.py

```diff
@@ -45,6 +45,9 @@ def validate(
     # <ReferralNFA> (N00112) = false or 0
     # then the <CINreferralDate> (N00100) for this module must be the latest of all Referral Dates for that child.
 
+    # removing nans prevents ValueError: attempt to get argmax of an empty sequence, when no CINreferralDate is present
+    df_cin = df_cin[df_cin[CINreferralDate].notna()]
```

Comment:
Good idea, I didn't think of this because I was worried about this meaning there could be multiple empty CInreferralDates, but re-reading the logic, that doesn't matter! Good fix!